packages/provider/src/transports/base-provider-transport.ts

```diff
@@ -222,14 +226,17 @@ export abstract class BaseProviderTransport implements ProviderTransport {
       }
     }
 
-    // ACCOUNTS_CHANGED -- when a user logs in or out
+    // ACCOUNTS_CHANGED -- when a user logs in, logs out or disconnects an app
     if (message.type === EventType.ACCOUNTS_CHANGED) {
-      this.accountsChangedPayload = undefined
-      if (message.data && message.data.length > 0) {
-        this.accountsChangedPayload = ethers.utils.getAddress(message.data[0])
-        this.events.emit('accountsChanged', [this.accountsChangedPayload])
+      this.accountsChangedPayload = { accounts: [] }
+      if (message.data && (message.data.length > 0 || message.data.accounts?.length > 0)) {
```

Comment:
Is it possible for `message.data.length === 0` when `message.data.accounts?.length > 0`?